src/v/serde/type_str.h

```diff
@@ -13,33 +13,21 @@
 
 namespace serde {
 
-#if defined(_MSC_VER)
-#define SERDE_SIG __FUNCSIG__
-#elif defined(__clang__) || defined(__GNUC__)
-#define SERDE_SIG __PRETTY_FUNCTION__
-#else
-#error unsupported compiler
-#endif
-
 template<typename T>
-constexpr std::string_view type_str() {
+consteval std::string_view type_str() {
 #if defined(__clang__)
     constexpr std::string_view prefix
       = "std::string_view serde::type_str() [T = ";
     constexpr std::string_view suffix = "]";
-#elif defined(_MSC_VER)
-    constexpr std::string_view prefix
-      = "class std::basic_string_view<char,struct std::char_traits<char> > "
-        "__cdecl serde::type_str<";
-    constexpr std::string_view suffix = ">(void)";
-#else
+#elif defined(__GNUC__)
     constexpr std::string_view prefix
       = "constexpr std::string_view serde::type_str() [with T = ";
```

Comment:
Would probably work also with `constexpr` here. But I think it makes sense to check the compiler output (value of `__PRETTY_FUNCTION__` and use exactly this string as prefix to remove.